consensus/blake3pow/consensus.go

```diff
@@ -344,6 +353,42 @@ func (blake3pow *Blake3pow) CalcDifficulty(chain consensus.ChainHeaderReader, pa
 	return parent.Difficulty()
 }
 
+func (blake3pow *Blake3pow) CalcEntropyThreshold(chain consensus.ChainHeaderReader, parent *types.Header) *big.Int {
+	// EntropyThreshold ET
+	// For a header n+1, ET_(n+1) = ET_n + k * (targetTime*(s_n - s_(n-1))/(t_n - t_(n-1)) - ET_n)
+
+	nodeCtx := common.NodeLocation.Context()
+	if nodeCtx == common.ZONE_CTX {
+		log.Error("CalcEntropyThreshold cannot be called in a zone chain")
+		return nil
+	}
+
+	x := new(big.Int)
+	t := new(big.Int)
+
+	if parent.Hash() == chain.Config().GenesisHash {
+		return parent.EntropyThreshold()
+	}
+	parentOfParent := chain.GetHeaderByHash(parent.ParentHash())
+
+	bigTime := new(big.Int).SetUint64(parent.Time())
+	bigParentTime := new(big.Int).SetUint64(parentOfParent.Time())
+	targetTime := blake3pow.config.DurationLimit[nodeCtx]
+
+	x = x.Sub(parent.CalcS(), parent.ParentEntropy())
```

Comment:
I believe the threshold should be based on each zone's prior coincident entropy, not the simple parent entropy.

Reply:
According to the team behind Quai Network, with the increasing demand, traditional Layer-1 projects fall short in some areas and have difficulty providing sufficient scalability to users. Quai Network, on the other hand, aims to be the first decentralized blockchain that can be scaled by offering an innovative and scalable blockchain solution at this point.